### ISSUE TITLE
Make Gemini create its own DB tables

### ISSUE THREAD

**Issue by jonathangreen** [Islandora-CLAW/CLAW#722]
For the most part when I'm installing a web app, I expect to be able to point it at an empty DB, and it till do what it needs and create its own tables. This is not the case with Gemini, and it makes supporting Postgres and MySQL difficult, and makes the playbook to install longer then I think it needs to be. 

I'd like to see if there is a good way to have Gemini handle the table creation internally.

Thoughts anyone?

**Comment by ajs6f** [comment]
Both MySQL and Postgres have `CREATE IF NOT EXISTS`, but I dunno about other dbs.

**Comment by dannylamb** [comment]
@jonathangreen Composer script to run after deployment?

**Comment by jonathangreen** [comment]
The biggest issue I had was the `ID` column. Postgres wants `SERIAL` and MySQL wants `AUTOINCREMENT`. I was going to see what our database layer has for DB schema support. Maybe we can do something clever within the app.

**Comment by dannylamb** [comment]
@jonathangreen DBAL looks like it has cross-db tools for checking and creating tables: http://docs.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/schema-manager.html

**Comment by jonathangreen** [comment]
@dannylamb yes! something like that was exactly what I was thinking. How do you feel about gemini managing its schema using schema-manager?

**Comment by dannylamb** [comment]
:+1: Sounds good to me.

**Comment by DiegoPino** [comment]
@seth-shaw-unlvI will make https://github.com/Islandora-CLAW/Crayfish/blob/master/Gemini/src/Migrations/Version20180530031926.php#L13 aware of postgress instead of having 2x migration files. Too complicated without access to modify app['db'] directly from Crayfish (part of commons).

Again, something like https://github.com/Islandora-CLAW/CLAW/issues/828#issuecomment-393595733

And will integrate into composer.json script section. Wonder if you want this to run on each install or on each update?

Also: I think we need to have an extra "migrate your data if you already have stuff there" option. Probably worth another ticket.

**Comment by seth-shaw-unlv** [comment]
I've created a new Crayfish branch (https://github.com/Islandora-CLAW/Crayfish/tree/issue-828) with the postgresql changes (took a while to get that right) but I'll hold off on the PR until we have the updated playbook. I'm working on that right now with a local ansible-role-crayfish branch. 

I'm trying to get ansible to run `composer install` and `php bin/console migrations:migrate`. Hopefully I'll get it working soon.
